universal-application-tool-0.0.1/app/models/Application.java

```diff
@@ -11,9 +11,19 @@
 import play.data.validation.Constraints;
 import services.applicant.ApplicantData;
 
+/**
+ * An EBean mapped class that records the submission of a single applicant to a single program.
+ *
+ * <p>When a resident or trusted intermediary submits an application for a program, an {@code
+ * Application} is created with a snapshot of the resident's {@code ApplicantData}. This ensures
+ * that the data seen by the program administrator is consistent with what was actually submitted
+ * and not subject to change if the resident or trusted intermediary changes answers to shared
+ * questions after submitting.
```

Comment:
You might also note that a draft application is created when an applicant submits at least one block for a program, and the application transitions to `active` when submitted from the review page, and `obsolete` when a second application is submitted for the same program.

Reply:
Done